### PR TITLE
Add publisher container

### PR DESCRIPTION
**Pull request by pnuu**
This PR adds a threaded container to hold an instance of NoisyPublisher and tests for both ListenerContainer and PublisherContainer.

### CODE REVIEW

**Comment by mraspaud** [comment]
Thanks ! can you give an example in the documentation on how to use it ?

**Comment by pnuu** [comment]
Usage examples added for both ListenerContainer and PublisherContainer. Also fixed the indentation of the pre-existing examples.

File: posttroll/publisher.py [hunk 1]
```diff
@@ -262,3 +262,41 @@ def __enter__(self):
...
+        """Create a publisher instance"""
+        if self.pub is None:
+            self.pub = NoisyPublisher(name, port=port, aliases=aliases,
+                                      broadcast_interval=broadcast_interval,
+                                      nameservers=nameservers)
+
+    def send(self, msg):
+        """Send message"""
+        self.pub.send(str(msg))
+        self.logger.info("Sent message: %s", str(msg))
```
Comment (mraspaud) [inline posttroll/publisher.py]: Is this thread safe ? what's the difference then with NoisyPublisher ? Because there is no `run` method implemented...

Comment (pnuu) [inline posttroll/publisher.py]: The idea was to have NoisyPublisher running all the time in another thread and only pass the message without starting the publisher every time or wrapping long stretches of code inside the Publish context.

Don't see what could cause thread safety issues as only a string is passed to NoisyPublisher thread.

**Comment by pnuu** [comment]
I removed the whole PublisherContainer as (Noisy)Publisher can do the job directly. This PR can be considered as providing one more unittest (for ListenerContainer) and fixing indentation in the documentation code examples.

**Pull request merged**